pages/studentPortal/labs/multiplication/game.tsx

```diff
@@ -16,13 +21,126 @@ import {
 export type GameBlockState = {
   text: string;
   value: number;
   isProduct: boolean;
   state: BlockState;
 };
 
+export function initializeGameState(level: gameLevel): GameBlockState[] {
+  let dummyArray: GameBlockState[] = [];
+  for (let i = 0; i <= 19; i++) {
+    if (level === gameLevel.BEGINNER) {
+      let x = getRndInteger(1, 10);
+      let y = getRndInteger(1, 10);
+      let product: number = x * y;
+      let productString: string = x + " x " + y;
+      let initiateBlockState: GameBlockState = {
+        text: product.toString(),
+        value: product,
+        isProduct: true,
+        state: BlockState.NOT_SELECTED,
+      };
+      dummyArray.push(initiateBlockState);
+
+      initiateBlockState = {
+        text: productString,
+        value: product,
+        isProduct: false,
+        state: BlockState.NOT_SELECTED,
+      };
+      dummyArray.push(initiateBlockState);
+    } else if (level === gameLevel.BEGINNER_ADVANCED) {
+      let x = getRndInteger(10, 20);
+      let y = getRndInteger(10, 20);
+      let product: number = x * y;
+      let productString: string = x + " x " + y;
+      let initiateBlockState: GameBlockState = {
+        text: product.toString(),
+        value: product,
+        isProduct: true,
+        state: BlockState.NOT_SELECTED,
+      };
+      dummyArray.push(initiateBlockState);
+
+      initiateBlockState = {
+        text: productString,
+        value: product,
+        isProduct: false,
+        state: BlockState.NOT_SELECTED,
+      };
+      dummyArray.push(initiateBlockState);
+    } else if (level === gameLevel.INTERMEDIATE) {
+      let x = getRndInteger(20, 30);
+      let y = getRndInteger(20, 30);
+      let product: number = x * y;
+      let productString: string = x + " x " + y;
+      let initiateBlockState: GameBlockState = {
+        text: product.toString(),
+        value: product,
+        isProduct: true,
+        state: BlockState.NOT_SELECTED,
+      };
+      dummyArray.push(initiateBlockState);
+
+      initiateBlockState = {
+        text: productString,
+        value: product,
+        isProduct: false,
+        state: BlockState.NOT_SELECTED,
+      };
+      dummyArray.push(initiateBlockState);
+    } else if (level === gameLevel.INTERMEDIATE_ADVANCED) {
+      let x = getRndInteger(30, 40);
+      let y = getRndInteger(30, 40);
+      let product: number = x * y;
+      let productString: string = x + " x " + y;
+      let initiateBlockState: GameBlockState = {
+        text: product.toString(),
+        value: product,
+        isProduct: true,
+        state: BlockState.NOT_SELECTED,
+      };
+      dummyArray.push(initiateBlockState);
+
+      initiateBlockState = {
+        text: productString,
+        value: product,
+        isProduct: false,
+        state: BlockState.NOT_SELECTED,
+      };
+      dummyArray.push(initiateBlockState);
+    } else if (level === gameLevel.EXPERT) {
+      let x = getRndInteger(40, 50);
```

Comment:
Clean up this function using a helper function

Reply:
will create another branch to handle this part of the pr